source/state_representation/src/AnalogIOState.cpp

```diff
@@ -0,0 +1,99 @@
+#include "state_representation/AnalogIOState.hpp"
+
+#include "state_representation/exceptions/IONotFoundException.hpp"
+
+using namespace state_representation::exceptions;
+
+namespace state_representation {
+
+AnalogIOState::AnalogIOState() : IOState<double>() {
+  this->set_type(StateType::ANALOG_IO_STATE);
+}
+
+AnalogIOState::AnalogIOState(const std::string& name, unsigned int nb_ios) : IOState<double>(name, nb_ios) {
+  this->set_type(StateType::ANALOG_IO_STATE);
+  this->data_ = Eigen::VectorXd::Zero(nb_ios);
+}
+
+AnalogIOState::AnalogIOState(const std::string& name, const std::vector<std::string>& io_names) :
+  AnalogIOState(name, io_names.size()) {
+  this->set_names(io_names);
+}
+
+AnalogIOState::AnalogIOState(const AnalogIOState& state) : AnalogIOState(state.get_name(), state.get_names()) {
+  if (state) {
+    this->set_data(state.data());
+  }
+}
+
+AnalogIOState AnalogIOState::Zero(const std::string& name, unsigned int nb_ios) {
+  AnalogIOState zero = AnalogIOState(name, nb_ios);
+  // as opposed to the constructor specify this state to be filled
```

Comment:
nitpick: the phrasing of the comment is a bit weird
```suggestion
  // specify that the default constructed zero state is non-empty
```

Reply:
This is legacy phrasing :smile: